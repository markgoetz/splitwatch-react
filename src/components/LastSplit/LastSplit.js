import React, { useContext } from 'react';
import renderTime from '../../lib/renderTime';
import TimerContext from '../../lib/TimerContext';

import styles from './LastSplit.module.css';

export default function LastSplit(props) {
    const context = useContext(TimerContext);

    if (context.splits.length > 0) {
        const split = context.splits[context.splits.length - 1];
        return <div className={styles.lastSplit}>Last split: { renderTime(split) }</div>;
    }
    
    return <div className={styles.lastSplit}>No splits recorded</div>;
}
