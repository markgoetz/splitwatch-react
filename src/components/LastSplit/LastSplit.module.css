.lastSplit {
  text-align: left;
  font-family: var(--font-sans);
  color: var(--color-grey-dark);
}