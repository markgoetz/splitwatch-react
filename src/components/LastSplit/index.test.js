import React from 'react';
import LastSplit from './LastSplit';
import { shallow } from 'preact-render-spy';

describe('LastSplit', () => {
  test('renders correctly with no splits', () => {
    const context = shallow(<LastSplit />);
    expect(context).toMatchSnapshot();
  });

  test('renders correctly with a split', () => {
    const context = shallow(<LastSplit split={300} />);
    expect(context).toMatchSnapshot();
  });
});