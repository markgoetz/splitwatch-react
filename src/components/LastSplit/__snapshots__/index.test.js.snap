// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastSplit renders correctly with a split 1`] = `
preact-render-spy (1 nodes)
-------
<div>Last split: 00:00:00.3</div>

`;

exports[`LastSplit renders correctly with no splits 1`] = `
preact-render-spy (1 nodes)
-------
<div>No splits recorded</div>

`;
